File: Python_Books/算法图解/grokking_algorithms_COPY/04_quicksort/c/02_recursive_sum.c
#include <stdio.h>

int sum(int *arr, int index, int size) {
	if (index == size)
		return 0;
	return arr[index] + sum(arr, index + 1, 4);
}

int main(void) {
	int arr[4] = { 1,2,3,4 };
	printf("%d", sum(arr, 0, 4));

	return 0;
}

File: Python_Books/算法图解/grokking_algorithms_COPY/04_quicksort/javascript/02_recursive_sum.js
const arr = [1, 2, 3, 4];

/**
 * Sums values in array recursively
 * @param {Array} arr Array of numbers
 * @return {number} Sum of the numbers
 */
const sumRecursive = ( arr ) => {
    if ( arr.length == 1 ) return arr[0];
    return arr[0] + sumRecursive( arr.slice( 1 ) );
};

console.log( sumRecursive( arr ) );  // 10
